es5-shim.js

```diff
@@ -1980,9 +1980,15 @@
         parseInt = (function (origParseInt) {
             var hexRegex = /^[\-+]?0[xX]/;
             return function parseInt(str, radix) {
-                var string = trim(str);
-                var defaultedRadix = $Number(radix) || (hexRegex.test(string) ? 16 : 10);
-                return origParseInt(string, defaultedRadix);
```

Comment:
Yes, this will work.
Also, this problem is for undefined and null only.

`Function.prototype.call.bind(String.prototype.trim)(undefined)`
throws Uncaught TypeError: String.prototype.trim called on null or undefined